lib/metasploit/framework/tcp/client.rb

```diff
@@ -80,6 +80,11 @@ def connect(global = true, opts={})
             dossl = ssl
           end
 
+        # For Postgres, always connect with SSL disabled; SSL is enabled after the initial connection is made
+        if defined?(self) && self.class.name =~ /Postgres/
```

Comment:
In general, lower level building blocks shouldn't know about higher level building blocks - is there maybe an alterantive implementation approach we could take to avoid changing `tcp/client.rb` ? 👀